### PR TITLE
[Workflow] Catch error when trying to get an uninitialized marking

### LINKED ISSUE #44213: Add an initialize method to workflows
### Description

At the moment the recommended way to initialize a workflow for an object `$subject` is to call `WorkflowInterface::getMarking($subject)`. Internally this will call the getter for the configured property holding the marking on `$subject`. If this returns null the configured initial marking will be set on `$subject`.  
The problem is that if you use typed properties in `$subject::class` the marking property has to be either nullable or has to initialize the marking property with the initial marking. In my opinion both of these options are suboptimal.  
Option 1 probably doesn't reflect your model in a good way, most of the times you don't want a marking to be null at any time.  
Option 2 conflicts with separation of concerns considerations.  
The only other fix I can think of is to implement the getter for your marking like this:
```PHP
public function getMarking(): string
{
    return $this->marking ?? null;
}
````
But I think this is very unintuitive and even worse than options 1 and 2.  

The only good solution I can think of is to add a method `WorkflowInterface::initialize($subject)` that will only set the marking property of `$subject` to the configured initial marking without trying to get the marking first.  
In my opinion this would also be a much more intuitive way of initializing a workflow than calling `WorkflowInterface::getMarking($subject)`.

If no one disagrees with my opinion and proposal on this I would gladly implement this feature myself.

### Example

```PHP
// src/Entity/Book
namespace App\Entity;

class Book
{
    private string $marking;

    public function getMarking(): string
    {
        return $this->marking;
    }

    public function setMarking(string $marking): void
    {
        $this->marking = $marking;
    }
}
```
```YAML
# config/workflow.yaml
framework:
    workflows:
        book:
            type: 'state_machine'
            marking_store:
                type: 'method'
                property: 'marking'
            supports:
                - App\Entity\Book
            initial_marking: on_shelf
            places:
                - on_shelf
                - ...
```
```PHP
// some code initializing the workflow in App\Entity\Book
$book = new Book();

// current way that results in an exception
/** @var WorkflowInterface $bookStateMachine */
$bookStateMachine->getMarking($book); // Typed property must not be accessed before initialization

// current way that works
$initialMarking = new Marking();
foreach ($bookStateMachine>getDefinition()->getInitialPlaces() as $initialPlace) {
    $initialMarking->mark($initialPlace);
}
$bookStateMachine->getMarkingStore()->setMarking($book, $initialMarking);
$book->getMarking(); // 'on_shelf'

// proposed way
$bookStateMachine->initialize($book);
$book->getMarking(); // 'on_shelf'
```

### PR DESCRIPTION
**Pull request by lyrixx**
| Q             | A
| ------------- | ---
| Branch?       | 4.4
| Bug fix?      | yes
| New feature?  | no
| Deprecations? | no
| Tickets       | Fix #44213
| License       | MIT
| Doc PR        |

### CODE REVIEW

File: src/Symfony/Component/Workflow/MarkingStore/MethodMarkingStore.php [hunk 1]
```diff
@@ -54,7 +54,14 @@ public function getMarking($subject): Marking
             throw new LogicException(sprintf('The method "%s::%s()" does not exist.', \get_class($subject), $method));
         }
 
-        $marking = $subject->{$method}();
+        $marking = null;
+        try {
+            $marking = $subject->{$method}();
+        } catch (\Error $e) {
+            if (!preg_match('/^Typed property ([\w\\\\@]+)::\$(\w+) must not be accessed before initialization$/', $e->getMessage())) {
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Workflow/MarkingStore/MethodMarkingStore.php]: shouldn't we also check that the class matches `get_debug_type($subject)`?

File: src/Symfony/Component/Workflow/Tests/MarkingStore/SubjectWithType.php [hunk 2]
```diff
@@ -0,0 +1,18 @@
+<?php
+
+namespace Symfony\Component\Workflow\Tests\MarkingStore;
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Workflow/Tests/MarkingStore/SubjectWithType.php:12]: missing license header (same in MethodMarkingStoreTest.php btw)

Comment (nicolas-grekas) [reply src/Symfony/Component/Workflow/Tests/MarkingStore/SubjectWithType.php:12]: and in many other files in the component! :scream: 
can you please fix them all in this PR while at it?

File: src/Symfony/Component/Workflow/MarkingStore/MethodMarkingStore.php (hunk 1 above)
Comment (lyrixx) [reply src/Symfony/Component/Workflow/MarkingStore/MethodMarkingStore.php]: Fixed

File: src/Symfony/Component/Workflow/Tests/MarkingStore/SubjectWithType.php (hunk 2 above)
Comment (lyrixx) [reply src/Symfony/Component/Workflow/Tests/MarkingStore/SubjectWithType.php:12]: Actually, many files (outside this component) misse the header. I'm gonna do another PR

Comment (lyrixx) [reply src/Symfony/Component/Workflow/Tests/MarkingStore/SubjectWithType.php:12]: https://github.com/symfony/symfony/pull/46018

**Review by nicolas-grekas — APPROVED** [review]

**Comment by nicolas-grekas** [comment]
Thank you @lyrixx.

File: src/Symfony/Component/Workflow/MarkingStore/MethodMarkingStore.php (hunk 1 above)
Comment (stof) [reply src/Symfony/Component/Workflow/MarkingStore/MethodMarkingStore.php]: what if the property is in a parent class ?

**Pull request merged**